service/start.go

```diff
@@ -119,6 +116,52 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+func (svc *service) startAllExistingAppsWalletSubscriptions(ctx context.Context, relay *nostr.Relay) {
+	var apps []db.App
+	result := svc.db.Where("wallet_pubkey IS NOT NIL").Find(&apps)
+	if result.Error != nil {
+		logger.Logger.WithError(result.Error).Error("Failed to fetch App records with non-empty WalletPubkey")
+		return
+	}
+
+	for _, app := range apps {
+		go func(app db.App) {
+			err := svc.startAppWalletSubscription(ctx, relay, *app.WalletPubkey, "")
+			if err != nil {
+				logger.Logger.WithError(err).WithFields(logrus.Fields{
+					"app_id": app.ID}).Error("Failed to subscribe to wallet")
+			}
+		}(app)
+	}
+}
+
+func (svc *service) startAppWalletSubscription(ctx context.Context, relay *nostr.Relay, appWalletPubKey string, appWalletPrivKey string) error {
+	var infoEventId string
+	if appWalletPrivKey != "" {
+		infoEvent, err := svc.GetNip47Service().PublishNip47Info(ctx, relay, appWalletPubKey, appWalletPrivKey, svc.lnClient)
+		if err != nil {
+			logger.Logger.WithError(err).Error("Could not publish NIP47 info")
+		}
+		infoEventId = infoEvent.ID
+	}
+
+	logger.Logger.Info("Subscribing to events for wallet ", appWalletPubKey)
+	sub, err := relay.Subscribe(ctx, svc.createFilters(appWalletPubKey))
+	if err != nil {
+		logger.Logger.WithError(err).Error("Failed to subscribe to events")
+	}
+
+	// register a subscriber for "app_deleted" events, which handles nostr subscription cancel and nip47 info event deletion
+	svc.eventPublisher.RegisterSubscriber(&deleteAppConsumer{nostrSubscription: sub, walletPubkey: appWalletPubKey, svc: svc, relay: relay, infoEventId: infoEventId})
```

Comment:
Doesn't this need to be only done for new apps?

Reply:
Yes, but maybe we would want to cancel subscription and delete nip47 event once every legacy app has been deleted? Than we would need to handle that in the `ConsumeEvent` doing some db query count for `wallet_pubkey IS NULL` ?
Also, we can check if there are any legacy apps in the hub at start and do not subscribe nor create the nip47 event for the master key.
Let me know what you think, and I can add code for that.

Reply:
I was wrong, the `if s.walletPubkey == walletPubKey {` check will always return false for legacy apps. Thanks!